src/main.rs

```diff
@@ -75,15 +74,24 @@ fn main() -> Result<()> {
 
     let logger = slog::Logger::root(drain, slog::o!());
 
+    let worker_count = env_usize("NSNCD_WORKER_COUNT", 8);
+    let handoff_timeout = Duration::from_secs(env_usize("NSNCD_HANDOFF_TIMEOUT", 3) as u64);
```

Comment:
3 seconds is way too low, at cachix.org we've been experiencing problems due to this. Can we make it 10?

Reply:
Are you able to tune the timeout with NSNCD_HANDOFF_TIMEOUT?

Reply:
Yes, however I'd much prefer a fix upstream.

https://github.com/NixOS/nixpkgs/pull/290355

Reply:
Can you open an issue where we can discuss this? Tell us more about what kind of load you're putting on nsncd at Cachix, how often you see the timeouts and on which calls, and what your upstream NSS calls are (e.g. are you reaching out to LDAP or some other networked service).